app/controllers/api/v1/donations_controller.rb

```diff
@@ -14,7 +14,7 @@ def index
   # Creates a new donation resource
   # /api/:version/users/:user_id/locations/:location_id/donations
   def create
-    donation = Donation.new(params.permit(:location_id, :user_id, :size, :comments))
+    donation = current_user.donation.new(params.permit(:location_id, :size, :comments))
```

Comment:
We might need a `rescue_from ActiveRecord::InvalidForeignKey`. This will be thrown if we try to save a donation with a `location_id` that does not exist in the `locations` table.